src/main/java/tigase/xmpp/impl/PresenceState.java

```diff
@@ -751,7 +746,7 @@ protected void processInProbe(Packet packet, XMPPResourceConnection session, Que
 					Element pres = conn.getPresence();
 
 					if (pres != null) {
-						sendPresence(null, null, packet.getStanzaFrom(), results, pres);
+						sendPresence(null, conn.getJID(), packet.getStanzaFrom(), results, pres);
```

Comment:
Is it possible that we could send `presence` with an error of `unauthorized` instead of sending `presence` with `type=unavailable`? I do think so, when session is logged out but packets are still being processed. Using `packet.getStanzaTo()` could be better, for packets incoming not from the user to which this session belongs.

Reply:
I'm not sure it would cause such effect. In case of session being terminated while packet being processed this would result in `NotAuthorizedException` being thrown which would simply be handled in `process()` method and it would be ignored (see `catch` statement: https://github.com/tigase/tigase-server/blob/76ed9cacd24582a0fca662478b5c41f80d449e0c/src/main/java/tigase/xmpp/impl/PresenceState.java#L437-L437)

Reply:
Then I think this change would be OK.